File: 2244-Minimum-Rounds-to-Complete-All-Tasks/2244-Minimum-Rounds-to-Complete-All-Tasks.py
class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        map = {}
        for i in tasks:
            if i in map:
                map[i] += 1
            else:
                map[i] = 1
        
        if 1 in map.values():
            return -1
        rounds = 0
        for i in map.keys():
            while(map[i] != 0 ):
                if map[i] == 4:
                    rounds += 2
                    map[i] = 0
                    break
                if map[i] == 2 or map[i] == 3:
                    rounds += 1
                    map[i] = 0
                    break

                map[i] -= 3
                rounds += 1
        return rounds

File: 2244-Minimum-Rounds-to-Complete-All-Tasks/2244-Minimum-Rounds-to-Complete-All-Tasks.java
class Solution {
    public int minimumRounds(int[] tasks) {
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i = 0; i < tasks.length; i++){
            if(map.containsKey(tasks[i])){
                map.put(tasks[i], map.get(tasks[i]) + 1);
            } else {
                map.put(tasks[i], 1);
            }
        }
        if(map.containsValue(1)){
            return -1;
        }
        int rounds = 0;
        for (Map.Entry<Integer, Integer> e : map.entrySet()) {
            while(e.getKey() != 0){
                if( e.getValue() == 4 ){
                    rounds += 2;
                    map.put(e.getKey(), 0);
                    break;
                }
                if( e.getValue() == 2 || e.getValue() == 3 ){
                    rounds++;
                    map.put(e.getKey(), 0);
                    break;
                }
                
                map.put(e.getKey(), e.getValue() - 3);
                rounds++;
            } 
        }
        return rounds;
    }
}